### DISCUSSION TITLE
Wait for multiple request to finish

### DISCUSSION THREAD

**Discussion by agustinzamar** [postmanlabs/postman-flows#105] [Ideas]
I think it is a common use case, where we need many requests to be finished before continuing with the flow. Right now I think the only way is to run them sequentially.

Please see this image as a representation of the use case
![Snag_202d04f0](https://user-images.githubusercontent.com/36794905/140388017-4ddeec14-6599-4616-88c4-f7b70a8d9a05.png)

Maybe a more complex approach can be having different types of waiting blocks, like in javascript we have Promise.all(), Promise.race(), etc

BTW: If there is some workaround for the use case I presented on the picture please let me know

**Comment by dev-kanishk** [comment]
Hey, @agustinz97 , will following flow help you to achieve your use-case,  Here the merge block will be executed only when it has received both of its inputs.
![Screenshot 2021-11-09 at 12 01 35 PM](https://user-images.githubusercontent.com/48204976/140873877-a24bebe3-fcf6-47c5-9f17-e058b1f18957.png)

**Reply by agustinzamar** [reply]
Hey! Thanks for the idea, I've implemented it and is working. The issue here is that it is not scalable because in my real flow I have 7 requests that I need to wait for completion.

**Reply by saswatds** [reply]
Hey, @agustinz97 there is another way you could use for synchronization. It involves using the Signal Ports. Here is how it would work if you do not need data from all 7 different requests, and just want to synchronize.
![Kapture 2021-11-30 at 16 04 00](https://user-images.githubusercontent.com/8796447/144031433-6f57b712-efcb-43eb-9d2a-eb73ea4f4eaa.gif)
